Source/WebCore/html/HTMLScriptElement.cpp

```diff
@@ -113,6 +114,45 @@ String HTMLScriptElement::crossOrigin() const
     return parseCORSSettingsAttribute(attributeWithoutSynchronization(crossoriginAttr));
 }
 
+ExceptionOr<void> HTMLScriptElement::setInnerTextForBinding(const std::variant<String, RefPtr<TrustedScript>>& value)
+{
+    ExceptionOr<String> stringValueHolder = WTF::switchOn(
+        value,
+        [this](const String& str) -> ExceptionOr<String> {
+            return trustedTypeCompliantString(TrustedType::TrustedScript, *scriptExecutionContext(), str, "HTMLElement innerText"_s);
+        },
+        [](const RefPtr<TrustedScript>& trustedScript) -> ExceptionOr<String> {
+            return trustedScript->toString();
+        }
+    );
+    if (stringValueHolder.hasException())
+        return stringValueHolder.releaseException();
+
+    auto stringValue = stringValueHolder.releaseReturnValue();
+
+    return setInnerText(WTFMove(stringValue));
+}
+
+ExceptionOr<void> HTMLScriptElement::setTextContentForBinding(const std::optional<std::variant<String, RefPtr<TrustedScript>>>& value)
+{
+    ExceptionOr<String> stringValueHolder = WTF::switchOn(
+        value.value_or(emptyString()),
+        [this](const String& str) -> ExceptionOr<String> {
+            return trustedTypeCompliantString(TrustedType::TrustedScript, *scriptExecutionContext(), str, "Node textContent"_s);
+        },
+        [](const RefPtr<TrustedScript>& trustedScript) -> ExceptionOr<String> {
+            return trustedScript->toString();
+        }
+    );
+    if (stringValueHolder.hasException())
+        return stringValueHolder.releaseException();
+
+    auto stringValue = stringValueHolder.releaseReturnValue();
+
+    setTextContent(WTFMove(stringValue));
```

Comment:
Should just do this in a single line and won’t need `WTFMove`.

Reply:
Ah yeah, this had an intermediate line originally but I'll change it to WTFMove in the follow up where it makes sense.